tfc_docs.cmake

```diff
@@ -30,3 +30,156 @@ function(tfc_add_example EXAMPLE_TARGET cpp_file)
     ${EXAMPLE_TARGET}
   )
 endfunction()
+
+function(tfc_add_docs FILE_PATH DOCS_DIR)
+  file(GLOB FILES_TO_COPY "${FILE_PATH}/*")
+  file(COPY ${FILES_TO_COPY} DESTINATION ${DOCS_DIR})
+endfunction()
+
+function(set_docs SPHINX_SOURCE ENABLE_DOXYGEN)
+  find_package(Doxygen REQUIRED)
```

Comment:
don't need to find doxygen if doxygen is not enabled right?

Reply:
Can this functionality be splitted

`tfc_docs_sphinx`
`tfc_docs_doxygen`

And remove the enable flag?

Reply:
that was my initial thought but this prevents that 

```cpp
  add_custom_command(OUTPUT ${SPHINX_INDEX_FILE}
    COMMAND
      ${SPHINX_EXECUTABLE} -b html
      # Tell Breathe where to find the Doxygen output
      -Dbreathe_projects.tfc=${DOXYGEN_OUTPUT_DIR}/xml
      ${SPHINX_SOURCE} ${SPHINX_BUILD}
    WORKING_DIRECTORY 
      ${CMAKE_CURRENT_BINARY_DIR}
    DEPENDS 
      venv_setup
      # Other docs files you want to track should go here (or in some variable)
      ${CMAKE_CURRENT_SOURCE_DIR}/index.rst
      # ${CMAKE_CURRENT_SOURCE_DIR}/design/cm.md
      ${DOXYGEN_INDEX_FILE}
    MAIN_DEPENDENCY 
      ${SPHINX_SOURCE}/conf.py
    COMMENT 
      "Generating documentation with Sphinx"
  )
```

because the doxygen index file is compiled with sphinx

Reply:
just tested, if I skip the index file doxygen doesn't work